.devcontainer/r-devel-clang-ubsan/Dockerfile

```diff
@@ -0,0 +1,23 @@
+FROM registry.gitlab.com/jangorecki/dockerfiles/r-devel-clang
+
+RUN apt-get -qq update \
+  && apt-get install -y --no-install-recommends git
+
+COPY DESCRIPTION .
+
+# install dependencies without ubsan flags
+RUN Rscript -e '                                          \ 
+read.dcf("DESCRIPTION", c("Imports", "Suggests")) |>      \
```

Comment:
{data.table} has very few deps so this is pretty manageable, only takes a few minutes to build.

But for ref. what we've done at [{lintr}](https://github.com/r-lib/lintr/pull/2459) is specify `Config/needs/development` with the _bare minimum_ set of packages you'd need to do reasonable development. The idea being if you particularly need vignettes now, you can just install {knitr} yourself from within the container.

Up to us whether we would exclude Suggests by default to get to development faster in the modal case.

Reply:
I recently pushed to GH related functionality. https://github.com/jangorecki/tools4pkgs/blob/b22666380ec6c675da539e75b49c9fa128718672/R/packages.R#L31